aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3ObjectMapper.java

```diff
@@ -44,24 +45,23 @@ public class S3ObjectMapper {
 
   private static final JsonFactory FACTORY = new JsonFactory();
   private static final ObjectMapper MAPPER = new ObjectMapper(FACTORY);
-  private static volatile boolean isInitialized = false;
+  private static final AtomicReference<ObjectMapper> REFERENCE = new AtomicReference<>();
 
   private S3ObjectMapper() {}
 
   static ObjectMapper mapper() {
-    if (!isInitialized) {
-      synchronized (S3ObjectMapper.class) {
-        if (!isInitialized) {
-          MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
-          MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-          MAPPER.setPropertyNamingStrategy(PropertyNamingStrategies.KebabCaseStrategy.INSTANCE);
-          MAPPER.registerModule(initModule());
-          isInitialized = true;
-        }
-      }
+    if (null == REFERENCE.get()) {
+      MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
```

Comment:
Trying to understand this change, arent multiple threads now going to conflict now when setting the MAPPER variable ?  There is still a single reference to MAPPER, isn't there?

Reply:
indeed, I've removed the single reference to `MAPPER` so that we don't conflict with multiple threads